src/groupeddataframe/groupeddataframe.jl

```diff
@@ -316,6 +316,12 @@ end
 
 Base.to_index(gd::GroupedDataFrame, key::Tuple) = gd.keymap[key]
 
+Base.to_index(gd::GroupedDataFrame, key::Pair) = 
+    Base.to_index(gd, (; Symbol(first(key)) => last(key)))
+
+Base.to_index(gd::GroupedDataFrame, key::Vector{Pair}) = 
```

Comment:
Current version will not work due to type invariance in Julia. Probably it is better to use:
```suggestion
Base.to_index(gd::GroupedDataFrame, key::Union{AbstractVector{<:Pair}, Base.Iterators.Pairs}) = 
```

Reply:
ah - maybe `Union` is not OK because of https://github.com/JuliaData/DataFrames.jl/pull/2265/files#diff-6349421a054bb7e74b7f27ae86304cbaR333 and current method dispatch rules.

Still then https://github.com/JuliaData/DataFrames.jl/pull/2265/files#diff-6349421a054bb7e74b7f27ae86304cbaR333 requires also an update to allow vector of vectors of pairs.